### PR TITLE
Add app apis

### PR DESCRIPTION
**Pull request by egs33**
Add support for App API

- [Get Apps](https://developer.kintone.io/hc/en-us/articles/115005336727)
- [Add Preview App](https://developer.kintone.io/hc/en-us/articles/115004712547)
- [Deploy App Settings](https://developer.kintone.io/hc/en-us/articles/115004881348)
- [Get App Deploy Status](https://developer.kintone.io/hc/en-us/articles/115004890947)
- [Get Form Fields](https://developer.kintone.io/hc/en-us/articles/115005509288)
- [Add Form Fields](https://developer.kintone.io/hc/en-us/articles/115005506868)
- [Update Form Fields](https://developer.kintone.io/hc/en-us/articles/115005507788)
- [Delete Form Fields](https://developer.kintone.io/hc/en-us/articles/115005341187)
- [Get Form Layout](https://developer.kintone.io/hc/en-us/articles/115005509068)
- [Update Form Layout](https://developer.kintone.io/hc/en-us/articles/115005341427)
- [Get Views](https://developer.kintone.io/hc/en-us/articles/115004401787)
- [Update Views](https://developer.kintone.io/hc/en-us/articles/115004880588)
- [Get General Settings](https://developer.kintone.io/hc/en-us/articles/115004811668)
- [Update General Settings](https://developer.kintone.io/hc/en-us/articles/115004868628)
- [Get Process Management Settings](https://developer.kintone.io/hc/en-us/articles/115005238087)
- [Update Process Management Settings](https://developer.kintone.io/hc/en-us/articles/115005066028)
- [Get Customization](https://developer.kintone.io/hc/en-us/articles/115004704967)
- [Update Customization](https://developer.kintone.io/hc/en-us/articles/115004873968)
- [Get App Permissions](https://developer.kintone.io/hc/en-us/articles/115005336327)
- [Update App Permissions](https://developer.kintone.io/hc/en-us/articles/115005445908)
- [Get Field Permissions](https://developer.kintone.io/hc/en-us/articles/115005362108)
- [Update Field Permissions](https://developer.kintone.io/hc/en-us/articles/115005190347)

### CODE REVIEW

File: src/kintone_client/app.cljc [hunk 1]
```diff
@@ -21,3 +21,480 @@
...
+    :codes - The App Code.
+             sequence of string
+
+    :name - The App Name.
+            string
+
+    :ids - The App IDs.
+           sequence of integer
+
+    :space-ids - The Space ID of where the App resides in.
```
Comment (makinoshi) [inline src/kintone_client/app.cljc]: Fix for grammer.
The space id where the app resides.
or
The space id in which the app is.

File: src/kintone_client/app.cljc [hunk 2]
```diff
@@ -21,3 +21,480 @@
   [conn app]
   (let [url (pt/-url conn path/form)]
     (pt/-get conn url {:params {:app app}})))
+
+(defn get-apps
+  "Gets general information of multiple Apps, including the name, description, related Space, creator and updater information.
+
+  opts
+
+    :offset - The number of retrievals that will be skipped.
```
Comment (makinoshi) [inline src/kintone_client/app.cljc:30]: Write default value when the value is not assigned.

File: src/kintone_client/app.cljc [hunk 3]
```diff
@@ -21,3 +21,480 @@
...
+              integer
+
+    :limit - The number of Apps to retrieve.
```
Comment (makinoshi) [inline src/kintone_client/app.cljc:34]: Write default value when the value is not assigned.

File: src/kintone_client/app.cljc [hunk 4]
```diff
@@ -21,3 +21,480 @@
...
+
+(defn get-form-layout
+  "Gets form layout of an App.
+
+  app - The kintone app ID.
+        integer
+
+  opts
+
+    :is-preview? - Gets from pre-live settings."
```
Comment (makinoshi) [inline src/kintone_client/app.cljc]: I prefer `preview?` to `is-preview?` has `is` prefix. IMO, that the style does not have it fits for Clojure style.
Can you fix it and other APIs written below?

File: src/kintone_client/app.cljc [hunk 5]
```diff
@@ -21,3 +21,480 @@
...
+
+(defn get-form-fields
+  "Gets the list of fields and field settings of an App.
+
+  app - The kintone app ID.
+        integer
+
+  opts
+
+    :lang - The localized language to retrieve the data.
```
Comment (makinoshi) [inline src/kintone_client/app.cljc:102]: IMO, "The language code" fits here and I want you to write supported codes like this.
https://kintone.github.io/kintone-js-sdk/latest/reference/app/#getformfieldsparams

File: src/kintone_client/app.cljc [hunk 6]
```diff
@@ -21,3 +21,480 @@
...
+  [conn app {:keys [lang is-preview?]}]
+  (let [params (cond-> {:app app}
+                 (some? lang) (assoc :lang lang))
+        url (pt/-url conn (if is-preview?
+                            path/preview-form-fields
+                            path/form-fields))]
+    (pt/-get conn url {:params params})))
+
+(defn add-form-fields
+  "Adds fields to a form of an App..
```
Comment (makinoshi) [inline src/kintone_client/app.cljc]: Dot duplicates

File: src/kintone_client/app.cljc [hunk 7]
```diff
@@ -21,3 +21,480 @@
...
+        url (pt/-url conn path/preview-form-fields)]
+    (pt/-put conn url {:params params})))
+
+(defn delete-form-fields
+  "Deletes fields from a form of an App.
+
+  app - The kintone app ID.
+        integer
+
+  codes - The list of field codes of the fields to delete.
```
Comment (makinoshi) [inline src/kintone_client/app.cljc:174]: Nice change!

**Review by makinoshi — COMMENTED** [review]
Thank you for your attentive implementations.

Overall, I want you to write 
- Comments to the APIs that can't be used with API token
- Guide the timing to use `deploy-app-settings`

**Review by saitouena — APPROVED** [review]

**Pull request merged**